block_device.go

```diff
@@ -45,6 +52,14 @@ func (bd *BlockDevice) Write(p []byte) (int, error) {
 			return 0, err
 		}
 
+		// This will only succeed on UBI volumes, fail silently on SD/MMC
+		//
+		// This must be here becuase UBI will verify if data written
+		// matches the size argument when the file descriptor is closed.
+		syscall.Syscall(syscall.SYS_IOCTL, out.Fd(),
+			uintptr(unsafe.Pointer(UBI_IOCVOLUP)),
+			uintptr(unsafe.Pointer(&bd.ImageSize)))
```

Comment:
Yes, will write up an better explanation. Could probably be inspired from http://lxr.free-electrons.com/source/include/uapi/mtd/ubi-user.h#L70

The reason we are not checking error return code here is because this is an attempt to keep the code generic. This would probably be the only location where we need an:
```
if UBI_UPDATE {
    //do something
}

``` 

And I have been actively trying to avoid these and the Mender client is not currently aware of what type it is updating. The idea was also that this fails "silently" on an SD/MMC. If it fails on UBI the update will fail, not optimal probably and maybe not worth keeping it "generic" for this reason?